### PR TITLE
[VDO-5697] Standardize on VDO_SUCCESS in utility code

### PR DESCRIPTION
**Pull request by lorelei-sakai**
This series ingests eight of Mike's suggested patches, all around the theme of using VDO_SUCCESS instead of UDS_SUCCESS in the utility files. I have added some internal-only commits, propagating Mike's change to internal code, and I have also added two new commits, one centered of removing unused error codes and making VDO_SUCCESS available to UDS code, and one centered on completing the VDO_ASSERT transition that Mike only did for VDO files but not for indexer files.

This series is built on top of, and depends on, the commits in PR 104.

### CODE REVIEW

File: src/c++/uds/src/uds/errors.h [hunk 1]
```diff
@@ -21,7 +21,8 @@
 /* Valid status codes for internal UDS functions. */
 enum uds_status_codes {
 	/* Successful return */
-	UDS_SUCCESS = 0,
+	VDO_SUCCESS = 0,
```
Comment (bjohnsto) [inline src/c++/uds/src/uds/errors.h:24]: This feels weird to me. In my mind, I was assuming there was a common include that defined VDO_SUCCESS = 0 and then we would include that everywhere it was needed, including the uds ones. I know that sort of goes against how we have uds feed into vdo currently, but is that something we are looking at changing down the road?

File: src/c++/vdo/tests/DmKcopydFake_t1.c [hunk 2]
```diff
@@ -86,7 +86,7 @@ static void testDmKcopyd(block_count_t regionSize)
 
   // Verify that the original data has not been touched.
   char *buffer;
-  UDS_ASSERT_SUCCESS(vdo_allocate(VDO_BLOCK_SIZE * regionSize, char,
+  VDO_ASSERT_SUCCESS(vdo_allocate(VDO_BLOCK_SIZE * regionSize, char,
                                   "verification buffer", &buffer));
   VDO_ASSERT_SUCCESS(layer->reader(layer, 0, regionSize, buffer));
   UDS_ASSERT_EQUAL_BYTES(buffer, data, VDO_BLOCK_SIZE * regionSize);
```
Comment (bjohnsto) [inline src/c++/vdo/tests/DmKcopydFake_t1.c:92]: Are we planning to convert these types of asserts as well?

File: src/c++/uds/userLinux/tests/Permassert_t1.c [hunk 3]
```diff
@@ -99,15 +99,15 @@ static void checkNotFound(const char *str)
 /**********************************************************************/
 static void testAssertionSuccess(void)
 {
-  UDS_ASSERT_SUCCESS(ASSERT(true, "true"));
+  UDS_ASSERT_SUCCESS(VDO_ASSERT(true, "true"));
```
Comment (bjohnsto) [inline src/c++/uds/userLinux/tests/Permassert_t1.c:102]: I think this should be VDO_ASSERT_SUCCESS? Or are we only converting tests to just use VDO_ASSERT?

**Review by bjohnsto — COMMENTED** [review]
Looks good but I did have a couple questions / comments

File: src/c++/uds/kernelLinux/uds/funnel-requestqueue.c [hunk 4]
```diff
@@ -275,5 +275,5 @@ void uds_request_queue_finish(struct uds_request_queue *queue)
 
 	uds_free_funnel_queue(queue->main_queue);
```
[the remark this answers is not in the document]
Comment (lorelei-sakai) [reply src/c++/uds/kernelLinux/uds/funnel-requestqueue.c]: Yes, uds_free_funnel_queue is an example of a function that Mike converted that should not be converted in this commit. This particular method will get converted, but in a later commit that deals with all the funnel_queue functions.

File: src/c++/vdo/tests/DmKcopydFake_t1.c (hunk 2 above)
Comment (lorelei-sakai) [reply src/c++/vdo/tests/DmKcopydFake_t1.c:92]: Which type of assert are you talking about here?

File: src/c++/uds/userLinux/tests/Permassert_t1.c (hunk 3 above)
Comment (lorelei-sakai) [reply src/c++/uds/userLinux/tests/Permassert_t1.c:102]: This test is in the uds directory, so it doesn't have VDO_ASSERT_SUCCESS defined. As I mentioned in the commit message, I didn't think that was worth doing.

File: src/c++/vdo/tests/DmKcopydFake_t1.c (hunk 2 above)
Comment (bjohnsto) [reply src/c++/vdo/tests/DmKcopydFake_t1.c:92]: UDS_ASSERT_EQUAL_BYTES

File: src/c++/uds/src/uds/errors.h (hunk 1 above)
Comment (bjohnsto) [reply src/c++/uds/src/uds/errors.h:24]: I wasn't suggesting a single file. I assumed we might have a very common file somewhere. If thats errors.h, then I might consider moving VDO_SUCCESS = 0 out of the uds_status_codes enum and put it at the top of the file.

Comment (lorelei-sakai) [reply src/c++/uds/src/uds/errors.h:24]: Hm, could do.

Comment (bjohnsto) [reply src/c++/uds/src/uds/errors.h:24]: very

**Review by bjohnsto — APPROVED** [review]
Looks good

**Pull request merged**